File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.java
class Solution {
    public int findMin(int[] nums) {
        return findMin(nums, 0, nums.length - 1);
    }

    private int findMin(int[] nums, int ini, int fi) {
        if (nums[ini] <= nums[fi]) return nums[ini];

        int m = (ini + fi) / 2;
        if (nums[ini] <= nums[m]) {
            return findMin(nums, m+1, fi);
        } else { // nums[ini] > nums[m]
            return findMin(nums, ini, m);
        }

    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        map = {}
        for n in nums:
            map[n] = map.get(n, 0) + 1
        freq = [[] for _ in range(len(nums) + 1)]
        for n in map:
            freq[map[n]].append(n)
        res = []
        for i in range(len(freq) - 1, -1, -1):
            if freq[i]:
                res += freq[i]
                k -= len(freq[i])
                if k == 0:
                    break
        return res
